libs/forecasting/src/lib/ForecastingDashboard/Controls/TransformationsTable.tsx

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { DetailsList, SelectionMode, Stack, Text } from "@fluentui/react";
+import {
+  defaultModelAssessmentContext,
+  ModelAssessmentContext
+} from "@responsible-ai/core-ui";
+import { localization } from "@responsible-ai/localization";
+import React from "react";
+
+import { forecastingDashboardStyles } from "../ForecastingDashboard.styles";
+import {
+  isMultiplicationOrDivision,
+  Transformation
+} from "../Interfaces/Transformation";
+
+interface ITransformationsTableProps {
+  transformations: Map<string, Transformation>;
+}
+
+interface ITransformationsTableState {
+  rows: ITransformationRow[];
+}
+
+const stackTokens = {
+  childrenGap: "l1"
+};
+
+interface ITransformationRow {
+  key: string;
+  transformationName: string;
+  method: string;
+}
+
+export class TransformationsTable extends React.Component<
+  ITransformationsTableProps,
+  ITransformationsTableState
+> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  public constructor(props: ITransformationsTableProps) {
+    super(props);
+    this.state = { rows: this.calculateUpdatedTable() };
+  }
+
+  public componentDidUpdate(): void {
+    // Currently, transformations are not editable or deletable.
+    // If that changes in the future we will have to update these checks.
+    const nTransformations = this.props.transformations.size;
+    const prevTransformationNames = new Set(
+      this.state.rows.map((t) => t.transformationName)
+    );
+    const didUpdate =
+      prevTransformationNames.size !== nTransformations ||
+      nTransformations !==
+        [...this.props.transformations.keys()].filter((t) =>
+          prevTransformationNames.has(t)
+        ).length;
+
+    if (didUpdate) {
+      this.setState({ rows: this.calculateUpdatedTable() });
+    }
+  }
+
+  public render(): React.ReactNode {
+    if (
+      this.props.transformations.size === 0 ||
+      this.state === undefined ||
+      this.state.rows === undefined ||
+      this.state.rows.length === 0
+    ) {
+      return;
```

Comment:
Do we have any placeholder in the case there is no row in the table?

Reply:
Great question! No, because we just don't show the table component at all in that case.